### PR TITLE
Lock extract/JSONification

### PR DESCRIPTION
**Pull request by sshirokov**
This extracts the lock-fondling code into a class in `server/lib/lock` and renames the `lock_file.rb` file to `lock.rb` under app, since it handles the HTTP side of locking not anything directly to have to do with files. This addresses #15 

Internally, the new lock file is a JSON file at the same path, and the "lock status" is now considering a key inside the lock file. This stabs in the direction of making #16 possible.

cc/ @skalnik

### CODE REVIEW

File: server/app.rb [hunk 1]
```diff
@@ -69,7 +69,7 @@ def progress
     post '/print' do
       require_basic_auth
       if locked?
-        halt 423, locked? # halt_on_lock helper?
+        halt 423, lock_data(:json => true) # halt_on_lock helper?
```
Comment (skalnik) [inline server/app.rb]: Lets kill this comment

File: server/lib/lock.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    def lock!
+      update_lock :status => :locked
+    end
+
+    def unlock!
+      update_lock :status => :unlocked
+    end
+
+    ## I/O API
+    def update_lock(h={})
```
Comment (skalnik) [inline server/lib/lock.rb]: Why can't this variable be named `hash`? or `data`? :heart: me some clarity

Comment (sshirokov) [inline server/lib/lock.rb]: Because I never changed it :-1: my b

**Comment by skalnik** [comment]
Diggin' it, other than the few comments I made :grin: :+1:

**Pull request merged**